src/main/java/control/DetailViewController.java

```diff
@@ -405,34 +407,59 @@ public void activeBlockChanged() {
                     .setText(detailView.formatDouble(shotBlock.getPaddingBefore()));
                 ((DirectorDetailView) detailView).getPaddingAfterField()
                     .setText(detailView.formatDouble(shotBlock.getPaddingAfter()));
-                initDropDown(shotBlock);
+                activeBlock = shotBlock;
                 detailView.setVisible();
                 detailView.setVisible(true);
                 // Re-init the detail view with new data
                 manager.getRootPane().getRootHeaderArea().reInitHeaderBar(detailView);
                 this.reInitForDirectorBlock();
-
             }
         } else {
             detailView.resetDetails();
             detailView.setInvisible();
         }
     }
-    
+
     /**
-     * Initialize the drop down menu.
-     * @param shotBlock the shotBlock to do that for
+     * Initialize drop down menu.
      */
-    private void initDropDown(DirectorShotBlock shotBlock) {
-        Set<Integer> indices = shotBlock.getTimelineIndices();
-        ((DirectorDetailView) detailView).getSelectCamerasDropDown().getItems().clear();
-        manager.getScriptingProject().getCameras().forEach(camera -> {
-                ((DirectorDetailView) detailView).getSelectCamerasDropDown()
-                    .getItems().add(camera.getName());
-            });
-        indices.forEach(e -> {
-                ((DirectorDetailView) detailView).getSelectCamerasDropDown()
-                    .getCheckModel().check(e);
-            });
+    private void initDropDown() {
```

Comment:
Pretty long method with nested loops and ifs. Maybe think about splitting it or adding some comments